components/src/composables/useFormValidation.ts

```diff
@@ -55,7 +55,15 @@ export default function useFormValidation() {
   };
 
   const validate = () => {
-    return Promise.all(formState.fieldset.map(field => field.validate()))
+    return Promise.all(
+      formState.fieldset.map(field => {
+        if (!field.dirty.value) {
+          field.setDirty(true);
+          field.startWatcher();
```

Comment:
Need this change for supporting some cross validation scenarios. consider a form with 2 input fields, 1st is a dropdown and 2nd is a text input. Dropdown is required by default but text input is only required depending on the value of dropdown. Consider following scenarios.

1). At the first load both field will be untouched.
| Field          | Dirty | Invalid | Validator Running |
|----------------|-------|---------|-------------------|
| Dropdown Field | false | false   | false             |
| Text Field     | false | false   | false             |

2 - Select an option from dropdown. this will make 2nd text input required.
| Field          | Dirty | Invalid | Validator Running |
|----------------|-------|---------|-------------------|
| Dropdown Field | true | false   | true             |
| Text Field     | false | false   | false             |

3 - Submit form. this will trigger all validate functions of each fields, since dropdown is selected, no errors will come. but since text input is empty, required validation will be shown.
| Field          | Dirty | Invalid | Validator Running |
|----------------|-------|---------|-------------------|
| Dropdown Field | true | false   | true             |
| Text Field     | false | true   | false             |

4 - Unselect item from dropdown. this will remove required validation from text input field. but the previously shown required validation will still be shown under input field. since dropdown is required its validation will run and show.
| Field          | Dirty | Invalid | Validator Running |
|----------------|-------|---------|-------------------|
| Dropdown Field | true | true   | true             |
| Text Field     | false | true   | false             |

In above case, until you submit the form again and or you type in the text input field, the validation message from text input field won't get removed. this change is addressing that issue. after this change upon submitting the form, all fields will considered dirty.

Reply:
Got it. Thanks